### PR TITLE
feat(discord): better play/pause support

### PR DESCRIPTION
**Pull request by charjac**
related to #660 

better elapsed time in discord.
Its probably too hacky, i did not find any better way to do it, i really dont like this interval.
What do you think ?

### CODE REVIEW

**Comment by nukeop** [comment]
Would it be possible to create a redux subscriber that listens to play/pause/fast forward/rewind events, and tells the Discord part what to do? Or does Discord require refreshing the state every few seconds?

**Comment by charjac** [comment]
discord control the timer, the only thing we can do is refresh the start timestamp.
We could refresh it when paused, store the paused elapsed time and then refresh it with some basic math when played again.

**Comment by charjac** [comment]
The discord activity is refreshed:
 - when playing track change
 - when paused
 - when play again
 - when track-list is cleared
 - when a track is removed from a track-list of 1 track

**Comment by charjac** [comment]
I remove the interval thing, i think its not perfect but better like this

**Comment by charjac** [comment]
> Would it be possible to create a redux subscriber that listens to play/pause/fast forward/rewind events, and tells the Discord part what to do? Or does Discord require refreshing the state every few seconds?

We should definitely do this for discord and mpris event to avoid using ipcRenderer inside redux actions. 
Will try something in another PR.

**Comment by nukeop** [comment]
Great, this is very nice now, thanks for improving this.

File: packages/main/src/services/discord/index.ts [hunk 1]
```diff
@@ -10,29 +9,56 @@ import Logger, { $mainLogger } from '../logger';
...
+      this.activity.startTimestamp = this.pauseStart;
+      return this.sendActivity();
+    }
+  }
+
+  async play() {
+    if (this.isReady && this.activity) {
+      this.pausedTotal += Date.now() - this.pauseStart;
+      if (this.activity) {
+        this.activity.details = this.activity.details.substr(0, this.activity.details.length - 8);
```
Comment (nathanfranke) [inline packages/main/src/services/discord/index.ts:49]:
```suggestion
        this.activity.details = this.activity.details.substr(0, this.activity.details.length - 7);
```
This is badly programmed. Would be fixed after https://github.com/nukeop/nuclear/pull/761 which doesn't hard code the length of `\nPaused`

**Pull request merged**